### PR TITLE
Throw a better exception message when an error occurs while retrieving the discovery/JWKS document

### PR DESCRIPTION
**Pull request by kevinchalet**
Closes https://github.com/IdentityModel/IdentityModel.OidcClient/issues/37.

/cc @leastprivilege

### CODE REVIEW

File: src/Shared/ProviderInformation.cs [hunk 1]
```diff
@@ -119,5 +141,16 @@ public void Validate()
 
             return info;
         }
+
+        private static async Task<string> FormatErrorAsync(HttpResponseMessage response)
```
Comment (kevinchalet) [inline src/Shared/ProviderInformation.cs:145]: FYI, I stole it from https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs#L163-L170.

**Comment by leastprivilege** [comment]
Sorry was totally swamped. I will keep this open.

**Comment by kevinchalet** [comment]
No prob'. Don't hesitate to ping me if you want me to update/rebase it.

**Comment by leastprivilege** [comment]
thanks! Finally have some to work on this.

**Pull request merged**